src/composables/useArrowNavigation.js

```diff
@@ -31,16 +32,37 @@ import { onMounted, ref, unref } from 'vue'
  *
  * @param {import('vue').Ref | HTMLElement} listElementRef component ref to mount navigation
  * @param {import('vue').Ref} defaultElementRef component ref to return focus to // Vue component
+ * @param {string} selector native selector of elements to look for
  * @param {object} options navigation options
  * @param {boolean} [options.confirmEnter=false] flag to confirm Enter click
  */
-export function useArrowNavigation(listElementRef, defaultElementRef, options = { confirmEnter: false }) {
+export function useArrowNavigation(listElementRef, defaultElementRef, selector, options = { confirmEnter: false }) {
 	const listRef = ref(null)
 	const defaultRef = ref(null)
+
 	const itemElements = ref([])
+	const itemElementsIdMap = computed(() => itemElements.value.map(item => item.id))
+	const itemSelector = ref(selector)
+
 	const focusedIndex = ref(null)
 	const isConfirmationEnabled = ref(null)
 
+	// Set focused index according to selected element
+	const handleFocusEvent = (event) => {
+		const newIndex = itemElementsIdMap.value.findIndex(id => id === event.target?.id)
+
+		if (newIndex !== -1) {
+			focusedIndex.value = newIndex
+		}
+	}
```

Comment:
https://github.com/nextcloud/spreed/assets/84044328/2992f8e1-abc1-4881-ac45-a78e04651067

I am just clicking on the arrow down key and the navigation gets trapped between two items. I don't know exactly the reason but removing  `handleFocusEvent` solved it 🤔.

Reply:
Thanks for checking! I saw and fixed this bug for the joined conversations before, will check now for the rest

Reply:
> nativelyFocusElement is handled by arrow keys, focuses the element, and (surprisingly) triggers handleFocusEvent

because there is a focus event emitted to the `listElementRef` upon `.focus()`, I think.

Reply:
Duplicated mutation is fixed, as well as items trap, with adding ids to the missing ones